hologram_dashboard/fetcher/hologram_availability_fetcher.h

```diff
@@ -0,0 +1,150 @@
+#include <string>
+#include <unordered_map>
+#include <gflags/gflags.h>
+#include "fetcher/fetcher_lib/hologram_availability.pb.h"
+#include "fetcher/fetcher_lib/hologram_config.pb.h"
+
+DEFINE_string(chipper_batch_job_cell, "undefined", "Specifies the server on which chipper file location is stored");
+DEFINE_string(chipper_GDPR_batch_job_cell, "undefined", "Specifies the server on which chipperGDPR file location is stored");
+DEFINE_string(prime_meridian_batch_job_cell, "undefined", "Specifies the server on which prime meridian file location is stored");
+DEFINE_string(path)
+namespace wireless_android_play_analytics {
+    class HologramAvailabilityFetcher {
+    public:
+        /* 
+        * Requires: Nothing
+        * Modifies: Nothing
+        * Effects: Default constructor without any settings
+        */
+        HologramAvailabilityFetcher();
+
+        /* 
+        * Requires: kvick_folder_in includes exactly all the systems and has 
+        *           valid mappings
+        * Modifies: kvick_folder
+        * Effects: Non-default constructor to populate the mapping between
+        *          systems and their disk locations
+        */
+        HologramAvailabilityFetcher(std::unordered_map<std::string, std::string> kvick_folder);
+
+        /* 
+        * Requires: hologram_config_path is a valid path
+        * Modifies: hologram_configs
+        * Effects: Non-default constructor to populate configs of hologram
+        */
+        HologramAvailabilityFetcher(std::string hologram_config_path);
+
+        /* 
+        * Requires: kvick_folder_in includes exactly all the systems and has 
+        *           valid mappings, hologram_config_path is a valid path
+        * Modifies: kvick_folder, hologram_config
+        * Effects: Non-default constructor to populate the mapping between
+        *          systems and their disk locations and configs
+        */
+        HologramAvailabilityFetcher(std::string hologram_config_path,
+                                    std::unordered_map<std::string, std::string> kvick_folder_in);
+
+        /* 
+        * Requires: None
+        * Modifies: kvick_folder
+        * Effects: Parses gflag commands, and throws an error if the requires
+        *          flags are not set
+        */
+        void parseCommand();
+
+        /* 
+        * Requires: kvick_folder and hologram_configs are populated
+        * Modifies: data_sources
+        * Effects: Populate the status of every data source for the past 7 days
+        *          including today
+        */
+        void populateData();
+
+        /* 
+        * Requires: None
+        * Modifies: None
+        * Effects: Sends data_sources to database
+        */
+        void sendToDatabase();
+
+        /* 
+        * Requires: path is a valid path to a config file
+        * Modifies: hologram_configs
+        * Effects: populate information on configs of the hologram
+        */
+        void populateConfigs(std::string path);
+
```

Comment:
Simplify apis in this file as discussed offline?

Reply:
this is changed at a separate pr